jupyterhub/handlers/base.py

```diff
@@ -1202,7 +1202,7 @@ def non_get(self, user_name, user_path):
             and self.current_user
             and user_name == self.current_user.name
         ):
-            server_name = user_path.split('/', 1)[0]
+            server_name = user_path.split('/', 2)[1]
```

Comment:
`[1]` will fail when this is empty. Instead, let's strip the leading / and still get the first element:

```suggestion
            server_name = user_path.lstrip('/').split('/', 1)[0]
```